File: shared/types/payment.ts
export type PaymentTypeName = 'monthly' | 'single' | 'partial' | 'full';

export interface Payment {
  id: string;
  student_id: string;
  course_id: string;
  amount: string;
  payment_date: string;
  payment_type: PaymentTypeName;
  comment?: string;
  created_at: string;
}

File: backend/app/models/payment.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum


class PaymentType(str, Enum):
    MONTHLY = "monthly"
    SINGLE = "single"
    PARTIAL = "partial"
    FULL = "full"


class Payment(BaseModel):
    id: str
    student_id: str
    course_id: str
    amount: float
    payment_date: str
    payment_type: PaymentType = PaymentType.PARTIAL
    comment: Optional[str] = None
    created_at: str = ""
    user_id: str = ""
    updated_at: str = ""
